core/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/BalancedShardsAllocator.java

```diff
@@ -749,10 +753,10 @@ private boolean allocateUnassigned() {
                     } else if (logger.isTraceEnabled()) {
                         logger.trace("No Node found to assign shard [{}]", shard);
                     }
-                    unassigned.ignoreShard(shard);
+                    changed |= unassigned.ignoreShard(shard, Optional.ofNullable(decision));
```

Comment:
We should properly pass the right decision here (as we don't know which decision it was (NO or THROTTLE)). I think that the solution is `deciders.canAllocate(node, allocation)` as we can have only `decision == null` if throttledNodes have an entry for `node`.